presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -166,7 +179,26 @@ protected Boolean visitAtTimeZone(AtTimeZone node, Void context)
         @Override
         protected Boolean visitSubqueryExpression(SubqueryExpression node, Void context)
         {
-            return true;
+            return AstUtils.preOrder(node)
+                    .filter(columnReferences::containsKey)
+                    .map(Expression.class::cast)
+                    .allMatch(this::analyzeSubqueryReference);
+        }
+
+        private boolean analyzeSubqueryReference(Expression referenceExpression)
+        {
+            FieldId fieldId = requireNonNull(columnReferences.get(referenceExpression), "No FieldId for Expression");
+            if (!Objects.equals(fieldId.getRelationId(), sourceScope.getRelationType().getRelationId())) {
+                // Not our scope, not our responsibility
```

Comment:
Add a better comment. `fieldId` scope can either be subquery scope (we don't care), ORDER BY scope (that is fine) or one of `sourceScope` ancestor (constant basically). All cases are OK from `AggregationAnalyzer` perspective.